torch/onnx/symbolic_opset9.py

```diff
@@ -1613,26 +1613,39 @@ def eye(g, n, m, dtype=None, layout=None, device=None, pin_memory=False):
     tensor = zeros(g, shape, dtype, layout, device)
     return g.op("EyeLike", tensor)
 
-@parse_args('v', 'v', 'v', 'v', 'i')
-def slice(g, self, dim, start, end, step):
-    if step != 1:
-        _unimplemented("slice", "step!=1 is currently not supported")
-    if start.node().kind() != 'onnx::Constant' or \
-            end.node().kind() != 'onnx::Constant' or dim.node().kind() != 'onnx::Constant':
-        if sym_help._operator_export_type == torch.onnx.OperatorExportTypes.ONNX:
-            raise RuntimeError('Unsupported: ONNX export of Slice with dynamic inputs. DynamicSlice '
-                               'is a deprecated experimental op. Please use statically allocated '
-                               'variables or export to a higher opset version.')
+
+def slice(g, self, *args):
+    if len(args) == 4:
+        # aten::slice(Tensor self, int dim, int start, int end, int step) -> Tensor
+        dim, start, end, step = args
+        step = _parse_arg(step, 'i')
+        if step != 1:
+            raise RuntimeError("step!=1 is currently not supported")
+        if start.node().kind() != 'onnx::Constant' or \
+                end.node().kind() != 'onnx::Constant' or dim.node().kind() != 'onnx::Constant':
+            if sym_help._operator_export_type == torch.onnx.OperatorExportTypes.ONNX:
+                raise RuntimeError('Unsupported: ONNX export of Slice with dynamic inputs. DynamicSlice '
+                                   'is a deprecated experimental op. Please use statically allocated '
+                                   'variables or export to a higher opset version.')
+            else:
+                start_unsqueezed = g.op("Unsqueeze", start, axes_i=[0])
+                end_unsqueezed = g.op("Unsqueeze", end, axes_i=[0])
+                dim_unsqueezed = g.op("Unsqueeze", dim, axes_i=[0])
+                return g.op("DynamicSlice", self, start_unsqueezed, end_unsqueezed, dim_unsqueezed)
         else:
-            start_unsqueezed = g.op("Unsqueeze", start, axes_i=[0])
-            end_unsqueezed = g.op("Unsqueeze", end, axes_i=[0])
-            dim_unsqueezed = g.op("Unsqueeze", dim, axes_i=[0])
-            return g.op("DynamicSlice", self, start_unsqueezed, end_unsqueezed, dim_unsqueezed)
-    else:
+            start = _parse_arg(start, 'i')
+            end = _parse_arg(end, 'i')
+            dim = _parse_arg(dim, 'i')
+            return sym_help._slice_helper(g, self, axes=[dim], starts=[start], ends=[end])
+    elif len(args) == 3:
+        # aten::slice(t[] l, int start, int end, int step) -> t[]
```

Comment:
interesting, I am curious where this aten function is defined?

Reply:
it's defined in torch/csrc/jit/runtime/register_prim_ops.cpp: https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/runtime/register_prim_ops.cpp#L503
and it's calling listSlice function defined here: https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/runtime/register_ops_utils.cpp#L429